ct002/ct002.py

```
@@ -0,0 +1,387 @@
+            fields = message.split("|")[1:]
+            xor = 0
+            for b in data[:-2]:
+                xor ^= b
+            expected_checksum = f"{xor:02x}".encode("ascii")
+            actual_checksum = data[-2:]
+            logger.debug(
+                "CT002 request checksum mismatch (expected %s, got %s)",
+                expected_checksum,
+                actual_checksum,
+            )
+            return fields, None
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)